batcher/aligned-batcher/src/retry/batcher_retryables.rs

```diff
@@ -60,20 +88,151 @@ pub async fn user_balance_is_unlocked_retryable(
+            RetryError::Transient(e.to_string())
+        })
+}
+
+pub async fn create_new_task_retryable(
+    batch_merkle_root: [u8; 32],
+    batch_data_pointer: String,
+    proofs_submitters: Vec<Address>,
+    fee_params: CreateNewTaskFeeParams,
+    payment_service: &BatcherPaymentService,
+    payment_service_fallback: &BatcherPaymentService,
+) -> Result<TransactionReceipt, RetryError<BatcherError>> {
+    info!("Creating task for: 0x{}", hex::encode(batch_merkle_root));
+    let call_fallback;
+    let call = payment_service
+        .create_new_task(
+            batch_merkle_root,
+            batch_data_pointer.clone(),
+            proofs_submitters.clone(),
+            fee_params.fee_for_aggregator,
+            fee_params.fee_per_proof,
+            fee_params.respond_to_task_fee_limit,
+        )
+        .gas_price(fee_params.gas_price);
+
+    let pending_tx = match call.send().await {
+        Ok(pending_tx) => pending_tx,
+        Err(ContractError::Revert(err)) => {
+            // Since transaction was reverted, we don't want to retry with fallback.
+            warn!("Transaction reverted {:?}", err);
+            return Err(RetryError::Permanent(BatcherError::TransactionSendError));
+        }
+        _ => {
+            call_fallback = payment_service_fallback
+                .create_new_task(
+                    batch_merkle_root,
+                    batch_data_pointer,
+                    proofs_submitters,
+                    fee_params.fee_for_aggregator,
+                    fee_params.fee_per_proof,
+                    fee_params.respond_to_task_fee_limit,
+                )
+                .gas_price(fee_params.gas_price);
+            match call_fallback.send().await {
+                Ok(pending_tx) => pending_tx,
+                Err(ContractError::Revert(err)) => {
+                    warn!("Transaction reverted {:?}", err);
+                    return Err(RetryError::Permanent(BatcherError::TransactionSendError));
+                }
+                _ => return Err(RetryError::Transient(BatcherError::TransactionSendError)),
+            }
+        }
+    };
+
+    // timeout to prevent a deadlock while waiting for the transaction to be included in a block.
+    timeout(Duration::from_millis(BATCH_INCLUSION_DELAY), pending_tx)
+        .await
+        .map_err(|e| {
+            warn!("Error while waiting for batch inclusion: {e}");
+            RetryError::Permanent(BatcherError::ReceiptNotFoundError)
+        })?
+        .map_err(|e| {
+            warn!("Error while waiting for batch inclusion: {e}");
+            RetryError::Transient(BatcherError::TransactionSendError)
+        })?
+        .ok_or(RetryError::Permanent(BatcherError::ReceiptNotFoundError))
+}
+
+pub async fn cancel_create_new_task_retryable(
+    batcher_signer: &SignerMiddlewareT,
+    batcher_signer_fallback: &SignerMiddlewareT,
+    iteration: Arc<Mutex<usize>>,
+    previous_gas_price: Arc<Mutex<U256>>,
```

Comment:
maybe I'm not seeing something but do we really need the the `Arc<Mutex>`s?

Reply:
We need these to avoid Rust errors when attempting to pass a `&mut` into an async closure. We discussed this with @Oppen and decided to handle `Arc<Mutex>` only within the `retry_function`, removing them from this function.